apps/nextjs-website/src/components/organisms/Auth/SignUpForm.tsx

```diff
@@ -0,0 +1,342 @@
+  useEffect,
+  useState,
+} from 'react';
+
+interface SignUpFormProps {
+  username: string;
+  password: string;
+  confirmPassword: string;
+  firstName: string;
+  lastName: string;
+  company: string;
+  role: string;
+
+  setUsername: Dispatch<SetStateAction<string>>;
+  setPassword: Dispatch<SetStateAction<string>>;
+  setConfirmPassword: Dispatch<SetStateAction<string>>;
+  setFirstName: Dispatch<SetStateAction<string>>;
+  setLastName: Dispatch<SetStateAction<string>>;
+  setCompany: Dispatch<SetStateAction<string>>;
+  setRole: Dispatch<SetStateAction<string>>;
+
+  onSignUp: () => Promise<boolean>;
+}
+
+const SignUpForm = ({
+  username,
+  password,
+  confirmPassword,
+  firstName,
+  lastName,
+  company,
+  role,
+  setUsername,
+  setPassword,
+  setConfirmPassword,
+  setFirstName,
+  setLastName,
+  setCompany,
+  setRole,
+
+  onSignUp,
+}: SignUpFormProps) => {
+  const {
+    auth: { signUp },
+    shared,
+  } = translations;
+
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [isPasswordValid, setIsPasswordValid] = useState(false);
+  const [isPasswordDirty, setIsPasswordDirty] = useState(false);
+
+  const { authStatus } = useAuthenticator((context) => [context.authStatus]);
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
+
+  const handleMouseDownPassword = (event: MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  const handleClickShowConfirmPassword = () =>
+    setShowConfirmPassword((show) => !show);
+
+  const handleMouseDownConfirmPassword = (
+    event: MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
+
+  const emailValidators: ValidatorFunction[] = [
+    (value: string) => ({
+      valid: emailMatcher.test(value),
+      error: shared.emailFieldError,
+    }),
+  ];
+
+  const validatePassword = useCallback(() => {
+    setIsPasswordValid(passwordMatcher.test(password));
+  }, [password]);
+
+  useEffect(() => {
+    validatePassword();
+  }, [password, validatePassword]);
+
+  const onSignUpClick = useCallback(() => {
+    if (
+      !firstName ||
+      firstName?.trim().length === 0 ||
+      !lastName ||
+      lastName?.trim().length === 0 ||
+      !username ||
+      username?.trim().length === 0 ||
+      !password ||
+      password?.trim().length === 0 ||
+      !confirmPassword ||
+      confirmPassword?.trim().length === 0
+    ) {
+      return;
+    }
+
+    if (password !== confirmPassword) {
+      return;
+    }
+
+    onSignUp();
+  }, [confirmPassword, firstName, lastName, onSignUp, password, username]);
+
+  if (authStatus === 'authenticated') {
+    redirect('/');
+  }
+
+  return (
+    <Box component='section'>
+      <Card variant='outlined' elevation={8}>
+        <Grid container justifyContent='center'>
+          <Grid item xs={11}>
+            <Typography variant='h3' pt={8} mb={4} textAlign='center'>
```

Comment:
```suggestion
            <Typography variant='h4' pt={8} mb={4} textAlign='center'>
```
from Figma, this should be `h4`